### ISSUE TITLE
Wrong loss values and perplexity reported in word language model example

### ISSUE THREAD

**Issue by IssamLaradji** [pytorch/examples#734]
I ran the word language model example in [1] and I get negative loss and 0 perplexity. My guess it is because the `nn.NLLoss` is being used in line 104 instead of the `nn.CrossEntropyLoss`. Do you think that's the issue with the code?

[1] https://github.com/pytorch/examples/blob/master/word_language_model/main.py

**Comment by zhangguanheng66** [comment]
https://github.com/pytorch/examples/pull/732 fixes the error.
